src/dd/dd.rs

```diff
@@ -0,0 +1,679 @@
+        self.is_stdout = false;
+    }
+
+    pub fn set_flag(&mut self, flag: FlagT) {
+        self.flags |= flag;
+    }
+
+    pub fn get_flag(flag: &str) -> FlagT {
+        match flag {
+            "block" => Conf::BLOCK,
+            "unblock" => Conf::UNBLOCK,
+            "lcase" => Conf::LCASE,
+            "ucase" => Conf::UCASE,
+            "swab" => Conf::SWAB,
+            "noerror" => Conf::NOERROR,
+            "notrunc" => Conf::NOTRUNC,
+            "sync" => Conf::SYNC,
+            _ => Conf::INVALID_FLAG,
+        }
+    }
+
+    pub fn is_flag_valid(flag: &str) -> bool {
+        match flag {
+            "ibs" | "obs" | "bs" | "cbs" | "if" | "of" | "skip" | "seek" | "count" => true,
+            _ => false,
+        }
+    }
+
+    pub fn has_flag(&self, flag: FlagT) -> bool {
+        if self.flags & flag != 0 {
+            true
+        } else {
+            false
+        }
+    }
+
+    pub fn str_flags(&self) -> String {
+        let mut result = String::new();
+        if self.flags & Self::BLOCK != 0 {
+            result += "[block]";
+        }
+        if self.flags & Self::UNBLOCK != 0 {
+            result += "[unblock]";
+        }
+        if self.flags & Self::LCASE != 0 {
+            result += "[lcase]";
+        }
+        if self.flags & Self::UCASE != 0 {
+            result += "[ucase]";
+        }
+        if self.flags & Self::SWAB != 0 {
+            result += "[swab]";
+        }
+        if self.flags & Self::NOERROR != 0 {
+            result += "[noerror]";
+        }
+        if self.flags & Self::NOTRUNC != 0 {
+            result += "[notrunc]";
+        }
+        if self.flags & Self::SYNC != 0 {
+            result += "[sync]";
+        }
+        result
+    }
+
+    const BLOCK: FlagT = 0b00000001;
+    const UNBLOCK: FlagT = 0b00000010;
+    const LCASE: FlagT = 0b00000100;
+    const UCASE: FlagT = 0b00001000;
+    const SWAB: FlagT = 0b00010000;
+    const NOERROR: FlagT = 0b00100000;
+    const NOTRUNC: FlagT = 0b01000000;
+    const SYNC: FlagT = 0b10000000;
+    const INVALID_FLAG: FlagT = 0b1_00000000;
+}
+
+impl fmt::Display for Conf {
+    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+        write!(
+            f,
+            "if: {input_filename}\n\
+             of: {output_filename}\n\
+             ibs: {ibs}\n\
+             obs: {obs}\n\
+             cbs: {cbs}\n\
+             skip: {skip}\n\
+             seek: {seek}\n\
+             count: {count}\n\
+             flags: {flags}\n",
+            input_filename = self.inp_name,
+            output_filename = self.out_name,
+            ibs = self.inp_block_size,
+            obs = self.out_block_size,
+            cbs = self.cnv_block_size,
+            skip = self.skip,
+            seek = self.seek,
+            count = self.count,
+            flags = self.str_flags()
+        )
+    }
+}
+
+pub fn parse_args() -> Conf {
+    let mut cf = Conf::default();
+    for arg in env::args().skip(1) {
```

Comment:
This function should be like:
```rust
pub fn parse_args(args: Vec<String>) -> Conf {
    let mut cf = Conf::default();
    for arg in args.into_iter().skip(1) {
        // stuff
    }
    // stuff
}
```

You shouldn't use `env::args()` because the args given to `uumain()` have the first element removed if the executable was started using the multicall binary (the `uutils` binary).  `env::args()` will return the list of args with `uutils` still present.

Reply:
Changed.